### ISSUE TITLE
cim issues with -o 2 and -o 4

### ISSUE THREAD

**Issue by bereziat** [heimdali/heimdali#97]
In the following of #88 

```
echo 0 .5 1 | cim -f  -x 3 -o 2 | tpr
```

should produce:

```
0 0.5 1
```

but produces:

```
0 0.499992 0.999985
```

and

```
echo 0 .5 1 | cim -f  -x 3 -o 4 | tpr
```

should produce:

```
0 0.5 1
```

but produces:

```
0 0.5 0
```

**Comment by dfroger** [comment]
How `echo 0 .5 1 | cim -f  -x 3 -o 2 | tpr` can produce `0 0.5 1` without rounding `0.499992` at printing?

This code:

```
#include <stdio.h>

int main()
{
    double v = 0.5;
    double p = 65535.;
    unsigned short i = (unsigned short) (v*p);
    double o = (double) i / p;
    printf("%g\n", o);
}
```

outputs:

```
0.499992
```

**Comment by dfroger** [comment]
With original inrimage:

```
echo 0 .5 1 | cim -f  -x 3 -o 2 | tpr -c -f='%d'
```

produces:

```
0 32768 65535 
```

and:

```
echo 0 .5 1 | cim -f  -x 3 -o 2 | tpr -c -f='%g'
```

produce:

```
0 0.500008 1 
```

Formulae in the [doc](http://heimdali.readthedocs.org/en/develop/fixed_point.html) must be fixed.

**Comment by dfroger** [comment]
Ok, that's problems of rounding to superior int and formatting options, I can fix it.

**Comment by dfroger** [comment]
Fixed in 277c3b78e7135c8efea1a5515ea89fd4db53d534

**Comment by bereziat** [comment]
ok for me !!!
